apps/workspaces/views.py

```
@@ -241,19 +243,49 @@ def post(self, request, **kwargs):
                 if workspace.qbo_realm_id:
                     assert_valid(realm_id == workspace.qbo_realm_id,
                                  'Please choose the correct Quickbooks online account')
+
                 qbo_credentials = QBOCredential.objects.create(
                     refresh_token=refresh_token,
                     realm_id=realm_id,
                     workspace=workspace
                 )
+
+                try:
+                    qbo_connector = QBOConnector(qbo_credentials, workspace_id=kwargs['workspace_id'])
+
+                    company_info = qbo_connector.get_company_info()
+                    qbo_credentials.country = company_info['Country']
+                    qbo_credentials.company_name = company_info['CompanyName']
+                    qbo_credentials.save()
+
+                except Exception as exception:
+                    return Response(
+                        json.loads(exception.response),
+                        status=status.HTTP_400_BAD_REQUEST
```

we shouldn't return 400 here, we should silently ignore the error in this step